app/models/manageiq/providers/amazon/cloud_manager/event_parser.rb

```diff
@@ -1,4 +1,33 @@
 module ManageIQ::Providers::Amazon::CloudManager::EventParser
+  def self.event_to_hash(event, ems_id)
+    event_hash = {
+      :event_type => event["eventType"],
+      :source     => "AMAZON",
+      :full_data  => event,
+      :ems_id     => ems_id
+    }
+
+    unless %i(config cloud_watch_api cloud_watch_ec2 cloud_watch_alarm).include?(event["event_source"])
+      raise "Unsupported event source #{event["event_source"]}"
+    end
+
+    send("parse_#{event["event_source"]}_event!", event, event_hash)
```

Comment:
chilly bang bang. What is modified by this method?
It would be much easier to follow if you return the modified version and assign it.

```ruby
event_hash = send("parse_#{event["event_source"]}_event", event)
```